core/src/main/java/com/linecorp/armeria/common/ContentTooLargeException.java

```diff
@@ -23,20 +23,72 @@
 
     private static final long serialVersionUID = 4901614315474105954L;
 
+    private long transferred;
+    private long transferring;
+    private long maximum;
+
     private static final ContentTooLargeException INSTANCE = new ContentTooLargeException(false);
 
     /**
      * Returns a {@link ContentTooLargeException} which may be a singleton or a new instance, depending on
      * {@link Flags#verboseExceptionSampler()}'s decision.
+     * @deprecated use {@link #builder()} instead.
      */
     public static ContentTooLargeException get() {
         return Flags.verboseExceptionSampler().isSampled(ContentTooLargeException.class) ?
                new ContentTooLargeException() : INSTANCE;
     }
 
+    /**
+     * Returns a {@link ContentTooLargeExceptionBuilder} which may return a singleton or a new instance in its
+     * {@link ContentTooLargeExceptionBuilder#build()} method, depending on
+     * {@link Flags#verboseExceptionSampler()}'s decision.
+     */
+    public static ContentTooLargeExceptionBuilder builder() {
+        return new ContentTooLargeExceptionBuilder();
+    }
+
     private ContentTooLargeException() {}
 
     private ContentTooLargeException(@SuppressWarnings("unused") boolean dummy) {
         super(null, null, false, false);
     }
+
+    ContentTooLargeException(long transferred, long transferring, long maximum) {
+        super(String.format("content length too large: transferred(%d) + transferring(%d) > maximum(%d)",
+                            transferred, transferring, maximum));
+
+        this.transferred = transferred;
+        this.transferring = transferring;
+        this.maximum = maximum;
+    }
+
+    /**
+     * Returns how many bytes of the content have been transferred.
+     */
+    public long transferred() {
+        return transferred;
+    }
+
+    /**
+     * Returns the expected total number of bytes in the request or response.
+     */
+    public long transferring() {
+        return transferring;
+    }
+
+    /**
+     * Returns the maximum number of content bytes allowed.
+     */
+    public long maximum() {
+        return maximum;
+    }
+
+    @Override
+    public synchronized Throwable fillInStackTrace() {
+        if (Flags.verboseExceptionSampler().isSampled(getClass())) {
+            super.fillInStackTrace();
+        }
```

Comment:
We should never call `super.fillInStackTrace()` if a user called the constructor with `dummy`. Could we add `private final boolean neverSample` which is set to `true` if the constructor with `dummy` is called?